Add support for Gotify notifications

lukapeschke opened the pull request:
This is a follow-up to https://github.com/containrrr/watchtower/pull/346 , where @simskij proposed to add support gotify notifications to shoutrrr. 

I mostly used the watchtower PR and the microsoft teams service for implementation reference, but I'm not really sure about shoutrrr's implementation patterns, so a review would be nice :smile:

simskij commented on pkg/services/gotify/gotify_test.go:
```diff
@@ -0,0 +1,26 @@
+}
+
+var _ = Describe("the Gotify plugin", func() {
+	It("should work", func() {
```
Maybe this could be a little more descriptive? What we're actually testing is that the url builder works.

lukapeschke replied:
Completely agree. This was directly "stolen" from the teams part. I also added a few more tests.

simskij replied:
Awesome! Thanks!

simskij commented on pkg/services/gotify/gotify.go:
```diff
@@ -0,0 +1,95 @@
+}
+
+// Send a notification message to Microsoft Teams
+func (service *Service) Send(message string, params *types.Params) error {
```
Would it make more sense from a navigation point if this was placed at the top with the other struct function pointer?

lukapeschke replied:
I like to have my top-level function below the helper functions they call (C habit :) ) I'd prefer to keep it like this if you don't mind

simskij replied:
Sure. No worries!

simskij commented on pkg/services/gotify/gotify.go:
```diff
@@ -0,0 +1,95 @@
+	return err
+}
+
+func buildURL(config *Config) (string, error) {
```
Are there any other format requirements of the token? For example that it should be this and that length or only contain alphanumericals or whatever. If so, that would be nice to validate as well.

lukapeschke replied:
Added some checks, based on what I've found in the Gotify code

piksel commented on pkg/services/gotify/gotify.go:
```diff
@@ -0,0 +1,95 @@
+func buildURL(config *Config) (string, error) {
+	token := config.Token
+	if len(token) < 1 {
+		msg := fmt.Sprintf("invalid gotify token \"%s\"", token)
```
```suggestion
		return "", fmt.Errorf("invalid gotify token \"%s\"", token)
```

piksel commented on pkg/services/gotify/gotify.go:
```diff
@@ -0,0 +1,95 @@
+	token := config.Token
+	if len(token) < 1 {
+		msg := fmt.Sprintf("invalid gotify token \"%s\"", token)
+		return "", errors.New(msg)
```
```suggestion
```

piksel commented on pkg/services/gotify/gotify.go:
```diff
@@ -0,0 +1,95 @@
+	return title
+}
+
+// Send a notification message to Microsoft Teams
```
Copy-paste error? Should probably refer to this service :)

piksel commented on pkg/services/gotify/gotify_config.go:
```diff
@@ -0,0 +1,37 @@
+		Host:       config.Host,
+		Scheme:     Scheme,
+		ForceQuery: false,
+	}
```
```suggestion
		Path:	config.Token,
	}
```

lukapeschke commented:
Thanks for the reviews guys! Rebased the branch, added a few unit tests and did the changes you requested. I'm not familiar at all with gingko/gomega though, so please let me know if the test layout could be improved :)

simskij commented:
LGTM! Let's wait for @piksel before merging.
Thanks a lot for the awesome work!

lukapeschke commented:
Thanks @simskij ! Last edit were just two comments in which I left "teams" instead of "Gotify"

simskij commented:
Again: thank you for your awesome contribution 👍

simskij commented:
@all-contributors add @lukapeschke for code and docs
